src/service/standard-actions-impl.js

```diff
@@ -68,6 +68,21 @@ export class StandardActions {
     /** @const @private {!./viewport/viewport-impl.Viewport} */
     this.viewport_ = Services.viewportForDoc(ampdoc);
 
+    // A meta[name="amp-action-whitelist"] tag, if present, contains,
+    // in its content attribute, a whitelist of actions on the special AMP target.
+    if (this.ampdoc.getRootNode() && this.ampdoc.getRootNode().head) {
+      const meta =
+        this.ampdoc.getRootNode().head
+            .querySelector('meta[name="amp-action-whitelist"]');
+
+      // Cache the whitelist of allowed AMP actions (if provided).
+      if (meta) {
+        /** @const @private {!Array<string>} */
```

Comment:
Actually, given the current code this field should really be type-annotated as `!Array<string>|undefined` given that it's conditionally initialized. It might be better to extract this logic to a helper method, and use `null` as the default value.